test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -332,3 +488,152 @@ func checkData(t *testing.T, start time.Time, lineCount int) {
 	)
 	assert.NoError(t, err)
 }
+
+func ValidateEntity(t *testing.T, logGroup, logStream string, end *time.Time, expectedEntity expectedEntity) {
+	log.Printf("Validating entity for log group: %s, stream: %s", logGroup, logStream)
+
+	logGroupInfo, err := getLogGroup()
+	for _, lg := range logGroupInfo {
+		if *lg.LogGroupName == logGroup {
+			log.Println("Log group " + *lg.LogGroupName + " exists")
+			break
+		}
+	}
+	assert.NoError(t, err)
+	begin := end.Add(-sleepForExtendedFlush * 4)
+	log.Printf("Query start time is " + begin.String() + " and end time is " + end.String())
+	queryId, err := getLogQueryId(logGroup, &begin, end)
+	assert.NoError(t, err)
+	log.Printf("queryId is " + *queryId)
+	result, err := getQueryResult(queryId)
+	assert.NoError(t, err)
+	if !assert.NotZero(t, len(result)) {
+		return
+	}
+	requiredEntityFields := map[string]bool{
+		entityType:        false,
+		entityName:        false,
+		entityEnvironment: false,
+		entityPlatform:    false,
+		entityInstanceId:  false,
+	}
+	for _, field := range result[0] {
+		switch aws.ToString(field.Field) {
+		case entityType:
+			requiredEntityFields[entityType] = true
+			assert.Equal(t, expectedEntity.entityType, aws.ToString(field.Value))
+		case entityName:
+			requiredEntityFields[entityName] = true
+			assert.Equal(t, expectedEntity.name, aws.ToString(field.Value))
+		case entityEnvironment:
+			requiredEntityFields[entityEnvironment] = true
+			assert.Equal(t, expectedEntity.environment, aws.ToString(field.Value))
+		case entityPlatform:
+			requiredEntityFields[entityPlatform] = true
+			assert.Equal(t, expectedEntity.platformType, aws.ToString(field.Value))
+		case entityInstanceId:
+			requiredEntityFields[entityInstanceId] = true
+			assert.Equal(t, expectedEntity.instanceId, aws.ToString(field.Value))
+		}
+		fmt.Printf("%s: %s\n", aws.ToString(field.Field), aws.ToString(field.Value))
+	}
+	allEntityFieldsFound := true
+	for field, value := range requiredEntityFields {
+		if !value {
+			log.Printf("Missing required entity field: %s", field)
+			allEntityFieldsFound = false
+		}
+	}
+	assert.True(t, allEntityFieldsFound)
+}
+
+func getLogQueryId(logGroup string, since, until *time.Time) (*string, error) {
+	var queryId *string
+	params := &cloudwatchlogs.StartQueryInput{
+		QueryString:  aws.String("fields @message, @entity.KeyAttributes.Type, @entity.KeyAttributes.Name, @entity.KeyAttributes.Environment, @entity.Attributes.PlatformType, @entity.Attributes.EC2.InstanceId"),
```

Comment:
nit: make this a const

Reply:
addressed in commit [84cc3d3](https://github.com/aws/amazon-cloudwatch-agent-test/pull/433/commits/84cc3d39def9c67c5a0460105bd990fe712f031e#diff-90e695612350cf075a13140798e4d267b81f04e5d9258b705f170ab76499d804R53)